### PR TITLE
Add support for thread-local setlocale() and localeconv().

### PR DESCRIPTION
**Pull request by keyurdg**
This also fixes an icompatibility (with Zend) where printf doesn't respect the
locale.

### CODE REVIEW

**Comment by JoelMarcey** [comment]
Hi @keyurdg, we are going to have to revert this pull request because of some segfaults happening on Ubuntu 14.04. Not sure what in this diff is causing it, but it is the commit that started the problem.

See #4954

So I am re-opening this.

Out of curiosity, what system did you build and test this PR on?

cc: @jasone, @sgolemon, @rrh

**Comment by keyurdg** [comment]
Hi @JoelMarcey I tested it on Cent 6 and 7 and we are running this in Prod on Cent 7.

Do you have a core dump for the backtrace? I'll also try this on an Ubuntu 14.04 VM tomorrow.

**Comment by JoelMarcey** [comment]
All I have is what is in https://github.com/facebook/hhvm/issues/4954

@rrh -- have any more debug info that you can provide?

File: hphp/test/slow/ext_string/setlocale-multiple-calls.php [hunk 1]
```diff
@@ -0,0 +1,14 @@
+<?php
+
+test(LC_TIME, 'fr_FR');
```
Comment (glensc) [inline hphp/test/slow/ext_string/setlocale-multiple-calls.php]: maybe use utf-8 locales. looks better in github and in terminal :)

`fr_FR.UTF-8`

**Comment by sgolemon** [comment]
FYI Looks like all the side-issues with this diff have been worked out.  Should be landing it today after all the test runs have completed.  Thanks for all your patience!

**Comment by sgolemon** [comment]
3df8f4c239984df4d9b53c2b451f34218025f2f0 should take care of it.

**Comment by keyurdg** [comment]
Thanks much for shepherding this through @sgolemon!

**Pull request closed without merging**